### PR TITLE
Introduce node-size and master-size params in kubetest

### PR DESCRIPTION
**Pull request by shyamjvs**
Inspired from https://github.com/kubernetes/test-infra/pull/7709. This hopefully makes our configs cleaner.

/cc @wojtek-t @krzyzacy

### CODE REVIEW

File: kubetest/main.go [hunk 1]
```diff
@@ -654,6 +658,18 @@ func prepareGcp(o *options) error {
...
+		} else if o.gcpNodeSize != "" {
+			os.Setenv("NODE_SIZE", o.gcpNodeSize)
+		}
+
+		if masterSize := os.Getenv("MASTER_SIZE"); masterSize != "" {
+			log.Printf("Please use --gcp-master-size=%s (instead of deprecated MASTER_SIZE)", masterSize)
+		} else if o.gcpMasterSize != "" {
+			os.Setenv("MASTER_SIZE", o.gcpMasterSize)
+		}
+
```
Comment (krzyzacy) [inline kubetest/main.go]: you can combine this into the `migrateGcpEnvAndOptions` above

Comment (shyamjvs) [reply kubetest/main.go]: But that seems to be common for gce, gke, etc... While I think this makes sense only for GCE?

Comment (krzyzacy) [reply kubetest/main.go]: for gke I think this can be somehow combined with https://github.com/kubernetes/test-infra/blob/master/kubetest/gke.go#L49?

Comment (shyamjvs) [reply kubetest/main.go]: The reason why I didn't want to mess this up with GKE is, there we're using sth like a json config for the "shape" and also master-size is sth that isn't relevant for GKE.

Comment (krzyzacy) [reply kubetest/main.go]: ack, maybe you can reuse `util.MigrateOptions`

and another question is, is there any plan to eliminate the global env and use this flag in k/k?

Comment (shyamjvs) [reply kubetest/main.go]: That'd be a bigger change IIUC as our current cluster creation scripts in k/k are in bash (we probably should eventually move them to go).

**Comment by shyamjvs** [comment]
/hold
I'm making the change suggested above.

File: kubetest/main.go (hunk 1 above)
Comment (krzyzacy) [reply kubetest/main.go]: on a side note, I'll probably look into how to clean up/validate kubetest flags

Comment (shyamjvs) [reply kubetest/main.go]: Done.

**Comment by shyamjvs** [comment]
Made the change suggested above - PTAL.

/hold cancel

**Comment by wojtek-t** [comment]
LGTM

**Comment by krzyzacy** [comment]
gonna watch https://github.com/kubernetes/test-infra/pull/7689 for a bit, and then send another kubetest bump

**Pull request merged**